paimon-core/src/main/java/org/apache/paimon/utils/StatsCollectorFactories.java

```diff
@@ -47,7 +53,9 @@ public static SimpleColStatsCollector.Factory[] createStatsFactories(
                                     .noDefaultValue());
             if (fieldMode != null) {
                 modes[i] = SimpleColStatsCollector.from(fieldMode);
-            } else if (SpecialFields.isSystemField(field)) {
+            } else if (SpecialFields.isSystemField(field)
+                    || (options.thinMode()
+                            && keyNames.contains(SpecialFields.KEY_FIELD_PREFIX + field))) {
```

Comment:
Do we have test to cover this? Disable all stats & enable thin mode, and check key stats.